src/NServiceBus.Persistence.AzureTable/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -45,47 +46,39 @@ static string DecodeFrom64(string encodedData)
 
         public async Task Subscribe(Subscriber subscriber, MessageType messageType, ContextBag context, CancellationToken cancellationToken = default)
         {
-            var table = client.GetTableReference(subscriptionTableName);
+            var table = client.GetTableClient(subscriptionTableName);
 
             var subscription = new Subscription
             {
                 RowKey = EncodeTo64(subscriber.TransportAddress),
                 PartitionKey = messageType.ToString(),
                 EndpointName = subscriber.Endpoint,
-                ETag = "*"
+                ETag = ETag.All
             };
 
+            // TODO: StorageException is not an expected exception type, replaced by RequestFailedException
+            // https://learn.microsoft.com/en-us/dotnet/api/azure.data.tables.tableclient.upsertentityasync?view=azure-dotnet
+
             try
             {
-                var operation = TableOperation.InsertOrReplace(subscription);
-                await table.ExecuteAsync(operation, cancellationToken).ConfigureAwait(false);
+                await table.UpsertEntityAsync(subscription, TableUpdateMode.Replace, cancellationToken).ConfigureAwait(false);
             }
-            catch (StorageException ex)
+            catch (RequestFailedException ex) when (ex.Status == 409)
             {
-                if (ex.RequestInformation.HttpStatusCode != 409)
-                {
-                    throw;
-                }
+                throw;
             }
             ClearForMessageType(messageType);
         }
 
         public async Task Unsubscribe(Subscriber subscriber, MessageType messageType, ContextBag context, CancellationToken cancellationToken = default)
         {
-            var table = client.GetTableReference(subscriptionTableName);
+            var table = client.GetTableClient(subscriptionTableName);
             try
             {
-                var subscription = new Subscription
-                {
-                    RowKey = EncodeTo64(subscriber.TransportAddress),
-                    PartitionKey = messageType.ToString(),
-                    EndpointName = subscriber.Endpoint,
-                    ETag = "*"
-                };
-                var operation = TableOperation.Delete(subscription);
-                await table.ExecuteAsync(operation, cancellationToken).ConfigureAwait(false);
+                await table.DeleteEntityAsync(messageType.ToString(), EncodeTo64(subscriber.TransportAddress),
```

Comment:
here I would find it useful to have locals representing the rowKey and the partition key